near-sdk/src/events.rs

```diff
@@ -0,0 +1,77 @@
+use std::borrow::Cow;
+
+use crate::env;
+use crate::serde::Serialize;
+use crate::serde_json;
+
+/// Representation of an event formatted according to [NEP-297](https://github.com/near/NEPs/blob/master/neps/nep-0297.md).
+///
+/// `standard`, `version`, and `event` are required by the specification. The `data` field is optional
+/// and omitted during serialization when it is `None`.
+#[derive(Debug, Serialize)]
```

Comment:
It makes sense to also allow for deserialization for easy integration of `near-sdk` for non-contract usages:

```suggestion
#[near(serializers = [json])]
#[derive(Debug)]
```

Reply:
You can omit bounding `T: Deserialize`, since it would be added automatically by `#[derive(Deserialize)]` which `#[near(serializers = [json])]` expands to, too.

Having `T: Serialize` does make sense, since it shouldn't be possible to create NEP-297 events, which `data` cannot be serialized